src/AbstractVectors.fypp

```diff
@@ -362,6 +370,22 @@ contains
         return
     end subroutine copy_basis_${type[0]}$${kind}$
 
+    subroutine rand_basis_${type[0]}$${kind}$(X, ifnorm)
+        class(abstract_vector_${type[0]}$${kind}$), intent(inout) :: X(:)
+        logical, optional, intent(in) :: ifnorm
+        logical                       :: if_normalize
+        ! internal
+        integer :: i
+
+        if_normalize = optval(ifnorm, .true.)
+
+        do i = 1, size(X)
+            call X(i)%rand(ifnorm=ifnorm)
+        end do
```

Comment:
You don't need the `if_normalize` line (which you actually never use since the test is done inside the call `X(i)%rand(ifnorm).